control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2797,19 +2798,21 @@ func (r *HostedControlPlaneReconciler) reconcileDefaultSecurityGroup(ctx context
 		// Not AWS platform, skip
 		return nil
 	}
-	if hcp.Status.Platform != nil && hcp.Status.Platform.AWS != nil && hcp.Status.Platform.AWS.DefaultWorkerSecurityGroupID != "" {
-		// Security group has already been created, nothing to do
-		return nil
-	}
+
```

Comment:
this was here to prevent flooding API rate limits so remove it might have consequences

Reply:
why not short-circuit based on tags changes here?